app/Http/Controllers/NoteController.php

```diff
@@ -29,6 +30,40 @@ public function index(Request $request)
         ]);
     }
 
+    public function getNotesByUsername()
+    {
+        $user = Auth::user();
+
+        if (!$user) {
+            return response()->json(['message' => 'User not logged in.'], 401);
+        }
+
+        return response()->json([
+            'notes' => User::with('notes')->find($user->id)['notes'],
+        ]);
+    }
+
+    public function getNoteById($note)
+    {
+        $user = Auth::user();
+
+        if (!$user) {
+            return response()->json(['message' => 'User not logged in.'], 401);
+        }
+
+        $note = Note::find($note);
```

Comment:
Use `findOrFail` and Laravel will take care of the rest. You don't need next if then